gui_scripts/gui_helpers/general_helpers.py

```diff
@@ -202,3 +204,210 @@ def stop(self):
+    ):
+        """
+        Delete an item from the tree.
+
+        :param target_index:    Index of item to be deleted.
+        """
+        path = self.model.filePath(target_index)
+        is_directory = QtCore.QFileInfo(path).isDir()
+
+        if is_directory:
+            reply = QtWidgets.QMessageBox.question(self, "Delete Directory",
+                                                   f"Are you sure you want to delete the directory '{path}' and all its contents?",
+                                                   QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No,
+                                                   QtWidgets.QMessageBox.No)
+        else:
+            reply = QtWidgets.QMessageBox.question(self, "Delete File",
+                                                   f"Are you sure you want to delete the file '{path}'?",
+                                                   QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No,
+                                                   QtWidgets.QMessageBox.No)
+
+        if reply == QtWidgets.QMessageBox.Yes:
+            if is_directory:
+                QtCore.QDir(path).removeRecursively()
+            else:
+                QtCore.QFile.remove(path)
+            self.refresh_view()  # Refresh the model to reflect the deletion
+
+    def _delete(self):
+        """
+        Deletes an item after a paste operation
+        """
+        if self.is_directory:
+            QtCore.QDir(self.copied_path).removeRecursively()
+        else:
+            QtCore.QFile.remove(self.copied_path)
+        self.refresh_view()  # Refresh the model to reflect changes
+
+    def handle_context_menu(
+        self,
+        position: QtCore.QModelIndex
+    ):
+        """
+        Callback function to handle contextEvent signal. The context menu
+        created by this function is rooted at position.
+
+        :param position:    index of item that generated the signal.
+        """
+        index = self.indexAt(position)
+        if index.isValid():
+            menu_obj = QtWidgets.QMenu(self)
+
+            menu_obj.addSeparator()
+            copy_action = menu_obj.addAction("Copy")
+            cut_action = menu_obj.addAction("Cut")
+            paste_action = menu_obj.addAction("Paste")
+            menu_obj.addSeparator()
+            delete_action = menu_obj.addAction("Delete")
+            menu_obj.addSeparator()
+
+            action = menu_obj.exec_(self.viewport().mapToGlobal(position))
+
+            if action == copy_action:
+                self.copy_item(index)
+            elif action == cut_action:
+                self.copy_item(index, is_cut_operation=True)
+            elif action == paste_action:
+                self.paste_item(index)
+            elif action == delete_action:
+                self.delete_item(index)
+
+    def refresh_view(self):
+        """
+        Refreshes the view of the tree model. Necessary when certain copy or cut
```

Comment:
Also these functions too I’m curious if sphinx raises any errors or warnings

Reply:
This is fine in sphinx. I tested this on the docstring of config_scripts.setup_config.read_config